Assets/Scripts/Maze.cs

```diff
@@ -162,18 +171,39 @@ private void CreateCellHighlight(GameCell[,] cells, Vector3 cubeSize, int x, int
         cells[x, z].highlight.SetActive(false);
     }
 
-    private Exit CreateExit(ArrayList availableExits, float cubeWidth, GameCell[,] cells, PositionConverter positionConverter)
+    private Exit CreateExit(ArrayList availableExits, ArrayList destructibleCubes, float cubeWidth, GameCell[,] cells, PositionConverter positionConverter)
     {
-        int index = new System.Random().Next(0, availableExits.Count);
+        int index = rnd.Next(0, availableExits.Count);
         Vector2 exitPostion = (Vector2)availableExits[index];
+        destructibleCubes.Remove(exitPostion);
         mazeExit.transform.localScale = new Vector3(cubeWidth / 10, 1f, cubeWidth / 10);
         Exit exit = CreateGameObject(exitPostion.x, 0.01f, exitPostion.y, mazeExit, "Exit");
+        exit.gameObject.GetComponent<Spin>().SetSpeed(GameManager.instance.GetExitSpinSpeed());
         Vector2 position = positionConverter.ConvertScenePositionToBoard(exit.transform.localPosition);
         cells[(int)position.x, (int)position.y].finding = exit;
 
         return exit;
     }
 
+    private void CreateFindings(ArrayList destructibleCubes, float cubeWidth, GameCell[,] cells, PositionConverter positionConverter)
```

Comment:
Prawdopodobnie chcemy, żeby ta funkcja umiała tworzyć różnego rodzaju findingi, ale to wymaga większego systemu, tak żeby finding miał swoje podklasy odpowiedzialne za dodanie bomby, życia itp. Ale to będzie osobny duży task, tak myślę.